### PR TITLE
[llvm] update to llvm11

### PR DESCRIPTION
**Pull request by strega-nil**
Updatin' to LLVM 11!

Also, removes the old comma hack from the options thing.

### CODE REVIEW

**Comment by strega-nil** [comment]
cc @Neumann-A :)

**Comment by ras0219-msft** [comment]
```
root ➜ ~/workspace/vcpkg (dev/roschuma/azblob ✗) $ git grep "string(REPLACE \";\"" | cat
ports/aws-sdk-cpp/portfile.cmake:string(REPLACE ";" "\\\\\\\\\\;" BUILD_ONLY "${BUILD_ONLY}")
ports/boost-python/b2-options.cmake:string(REPLACE ";" "," BUILD_PYTHON_VERSIONS "${BUILD_PYTHON_VERSIONS}")
ports/freetype/pkgconfig.patch:     string(REPLACE ";" ", " PKG_CONFIG_REQUIRED_PRIVATE "${PKG_CONFIG_REQUIRED_PRIVATE}")
ports/llvm/portfile.cmake:string(REPLACE ";" "," LLVM_ENABLE_PROJECTS "${LLVM_ENABLE_PROJECTS}")
ports/llvm/portfile.cmake:string(REPLACE ";" "," LLVM_TARGETS_TO_BUILD "${LLVM_TARGETS_TO_BUILD}")
ports/msix/install-cmake.patch:     string(REPLACE ";" " " DELAYFLAGS "${DELAYFLAGS}")
ports/skia/portfile.cmake:        string(REPLACE ";" "\",\"" TEMP "${INPUT_}")
ports/tensorflow-cc/portfile.cmake:             string(REPLACE ";" "\n\t\t" libs_to_link "${libs_to_link}")
ports/zstd/0002-crosscompile-windows.patch:             string(REPLACE ";" " " ${flag_var} "${${flag_var}}")
```

Here are some potentially affected locations that we should check

**Comment by PhoebeHui** [comment]
cc @yurybura, who also has a PR https://github.com/microsoft/vcpkg/pull/13998 to upgrade llvm to 11.0.0.

File: ports/llvm/0001-fix-install-paths.patch [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+ # Generate a list of CMake library targets so that other CMake projects can
+ # link against them. LLVM calls its version of this file LLVMExports.cmake, but
+ # the usual CMake convention seems to be ${Project}Targets.cmake.
+-set(CLANG_INSTALL_PACKAGE_DIR lib${LLVM_LIBDIR_SUFFIX}/cmake/clang)
++set(CLANG_INSTALL_PACKAGE_DIR share/clang)
+ set(clang_cmake_builddir "${CMAKE_BINARY_DIR}/${CLANG_INSTALL_PACKAGE_DIR}")
+ 
+ # Keep this in sync with llvm/cmake/CMakeLists.txt!
+-set(LLVM_INSTALL_PACKAGE_DIR lib${LLVM_LIBDIR_SUFFIX}/cmake/llvm)
++set(LLVM_INSTALL_PACKAGE_DIR share/llvm)
```
Comment (Neumann-A) [inline ports/llvm/0001-fix-install-paths.patch:15]: Why patch this? This should be handled by `vcpkg_fixup_cmake_targets()`.

File: scripts/cmake/vcpkg_configure_cmake.cmake [hunk 2]
```diff
@@ -266,12 +266,12 @@ function(vcpkg_configure_cmake)
     endforeach()
 
     set(rel_command
-        ${CMAKE_COMMAND} ${_csc_SOURCE_PATH} ${_csc_OPTIONS} ${_csc_OPTIONS_RELEASE}
+        ${CMAKE_COMMAND} ${_csc_SOURCE_PATH} "${_csc_OPTIONS}" "${_csc_OPTIONS_RELEASE}"
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_configure_cmake.cmake:269]: why the extra " ? _csc_OPTIONS should be passed as a list not a single string.

Comment (strega-nil) [reply scripts/cmake/vcpkg_configure_cmake.cmake:269]: No, it shouldn't be, because of how `set` works.

**Comment by strega-nil** [comment]
Closed in favor of #13998

**Pull request closed without merging**